### DISCUSSION TITLE
Public url for server sandbox

### DISCUSSION THREAD

**Discussion by maksimr** [codesandbox/codesandbox-client#4968] [Q&A] [answered]
I can't find how to get a public server sandbox URL. In my case, I want to pass it to a webpack-dev-server for the proper work of WebSocket(Live reload).

Currently, I use [`hostname | awk '{split($0,a,"-"); print "https://"a[3]"."a[1]".codesandbox.io"}'`](https://codesandbox.io/s/webpack-react-4jqol?file=/package.json) to construct such URL and hope that it would not break in the future.

For example, Gitpod has a [CLI API for such things](https://www.gitpod.io/docs/command-line-interface/) `gp url`

**Comment by lbogdan** [comment, marked as answer]
Hey @maksimr ,

You can get the sandbox URL from the environment variable `SANDBOX_URL`, you can see an example of that here: https://codesandbox.io/s/hardcore-goodall-l0gge?file=/src/index.js .

Also, with `webpack-dev-server` you can just set [`devServer.disableHostCheck: true`](https://webpack.js.org/configuration/dev-server/#devserverdisablehostcheck) and not worry about the URL at all.

**Reply by maksimr** [reply]
@lbogdan thanks, that's what I want. `disable-host-check` solve a different problem. I need the URL because without it webpack tries to connect websocket to the localhost.

**Reply by lbogdan** [reply]
You can use this complete dev server webpack config:

```jsonc
{
  // ...
  devServer: {
    disableHostCheck: true,
    host: '0.0.0.0',
    port: 9000,
    sockPort: 'location',
  }
}
```

as you can see here: https://github.com/lbogdan/csb-tailwind/blob/65d86b6bf785cf8d69860c4a6069679f6b9980a5/webpack.config.js#L36-L40

**Reply by djejaquino** [reply]
`SANDBOX_URL` no longer works

**Reply by filipelima18** [reply]
@djejaquino - This was used in our SSE sandboxes, which have been deprecated. We now use Devboxes, which have another env var with the name `CODESANDBOX_HOST` and a value like `9qputt-$PORT.csb.app`. This can be used to generate the Devbox URL by replacing the port with the port of the app. Or it can be constructed with `$HOSTNAME-$PORT.$CSB_BASE_PREVIEW_HOST`.
